vms/platformvm/state/subnet_only_validator.go

```diff
@@ -292,3 +326,68 @@ func (d *subnetOnlyValidatorsDiff) putSubnetOnlyValidator(state Chain, sov Subne
 	}
 	return nil
 }
+
+type activeSubnetOnlyValidators struct {
+	lookup map[ids.ID]SubnetOnlyValidator
+	tree   *btree.BTreeG[SubnetOnlyValidator]
+}
+
+func newActiveSubnetOnlyValidators() *activeSubnetOnlyValidators {
+	return &activeSubnetOnlyValidators{
+		lookup: make(map[ids.ID]SubnetOnlyValidator),
+		tree:   btree.NewG(defaultTreeDegree, SubnetOnlyValidator.Less),
+	}
+}
+
+func (a *activeSubnetOnlyValidators) get(validationID ids.ID) (SubnetOnlyValidator, bool) {
+	sov, ok := a.lookup[validationID]
+	return sov, ok
+}
+
+func (a *activeSubnetOnlyValidators) put(sov SubnetOnlyValidator) {
+	a.lookup[sov.ValidationID] = sov
+	a.tree.ReplaceOrInsert(sov)
+}
+
+func (a *activeSubnetOnlyValidators) delete(validationID ids.ID) bool {
+	sov, ok := a.lookup[validationID]
+	if !ok {
+		return false
+	}
```

Comment:
If we want to keep it as a safeguard for deletions it's fine but IMO `delete` should not be used instead of `get`.

Reply:
Unless `delete` took in the full `SubnetOnlyValidator` struct, we need this check. This is because deleting from the `tree` requires the full struct, not just the ID.